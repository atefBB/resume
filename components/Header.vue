<template>
  <div class="container">
    <el-row>
      <el-col :span="12">
        <div class="header-left">
          <h1 id="name">
            {{ header.name ? header.name : "Anonymous" }}
          </h1>
          <h2 id="title">
            {{ header.title ? header.title : "Hacker" }}
          </h2>
        </div>
      </el-col>
      <el-col :span="12">
        <div style="header-right">
          <ul class="icons">
            <li v-if="header.github_username" id="icons-github">
              <a
                target="_blank"
                :href="'https://www.github.com/' + header.github_username"
              >
                <i class="fab fa-github" title="Github link" />
              </a>
            </li>
            <li v-if="header.linkedin_username" id="icons-linkedin">
              <a
                target="_blank"
                :href="
                  'https://www.linkedin.com/in/' + header.linkedin_username
                "
              >
                <i class="fab fa-linkedin" title="Linkedin link" />
              </a>
            </li>
            <li v-if="header.twitter_username" id="icons-twitter">
              <a
                target="_blank"
                :href="'https://twitter.com/' + header.twitter_username"
              >
                <i class="fab fa-twitter" title="Twitter link" />
              </a>
            </li>
            <li v-if="header.email" id="icons-email">
              <a target="_blank" :href="'mailto:' + header.email">
                <i class="fas fa-at" title="Email link" />
              </a>
            </li>
            <li v-if="header.phone" id="icons-phone">
              <a target="_blank" :href="'tel:' + header.phone">
                <i class="fas fa-phone-alt" title="Phone link" />
              </a>
            </li>
            <li v-if="header.skype_username" id="icons-skype">
              <a
                target="_blank"
                :href="'skype:' + header.skype_username + '?userinfo'"
              >
                <i class="fab fa-skype" title="Skype link" />
              </a>
            </li>
          </ul>
        </div>
      </el-col>
    </el-row>
  </div>
</template>

<script>
import header from "~/assets/yaml/header.yml"
import { safeLoad } from "js-yaml"

export default {
  props: [],
  data() {
    return {
      header: null
    }
  },
  created: function() {
    this.getContent()
  },
  methods: {
    getContent() {
      this.header = safeLoad(header)
    }
  }
}
</script>
