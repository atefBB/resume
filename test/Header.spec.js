import faker from "faker"
import { testHelper, mountHelper } from "./helpers"
import Header from "~/components/Header.vue"
jest.mock(`~/assets/yaml/header.yml`)

describe("Header", () => {
  testHelper(Header)

  describe("Matches randoms values", () => {
    const values = {
      header: {
        name: faker.name.firstName() + " " + faker.name.lastName(),
        title: faker.name.jobTitle(),
        github_username: faker.internet.userName(),
        linkedin_username: faker.internet.userName(),
        twitter_username: faker.internet.userName(),
        email: faker.internet.email(),
        phone: faker.phone.phoneNumberFormat(),
        skype_username: faker.internet.userName()
      }
    }
    const mockFn = jest.fn()
    const options = {
      methods: {
        getContent: mockFn
      }
    }
    const factory = mountHelper(Header)

    it("Should call the mocked function", () => {
      factory({ options, values })
      expect(mockFn).toHaveBeenCalled() //we use correctly the mock function
    })

    //name
    it("Should contain the name", () => {
      const wrapper = factory({ options, values })
      expect(wrapper.find("div.header-left #name").text()).toEqual(
        expect.stringContaining(values.header.name)
      )
    })
    it("Should contain the default name", () => {
      let localValues = { header: { ...values.header, name: undefined } }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.find("div.header-left #name").text()).toEqual(
        expect.stringContaining("Anonymous")
      )
    })

    //title
    it("Should contain the title", () => {
      const wrapper = factory({ options, values })
      expect(wrapper.find("div.header-left #title").text()).toEqual(
        expect.stringContaining(values.header.title)
      )
    })
    it("Should contain the default title", () => {
      let localValues = { header: { ...values.header, title: undefined } }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.find("div.header-left #title").text()).toEqual(
        expect.stringContaining("Hacker")
      )
    })

    //github_username
    it("Should contain the github_username", () => {
      const wrapper = factory({ options, values })
      expect(
        wrapper.find("ul.icons li#icons-github a").attributes("href")
      ).toEqual(expect.stringContaining(values.header.github_username))
    })
    it("Should NOT contain the github_username", () => {
      let localValues = {
        header: { ...values.header, github_username: undefined }
      }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.find("ul.icons li#icons-github").exists()).toBe(false)
    })

    //linkedin_username
    it("Should contain the linkedin_username", () => {
      const wrapper = factory({ options, values })
      expect(
        wrapper.find("ul.icons li#icons-linkedin a").attributes("href")
      ).toEqual(expect.stringContaining(values.header.linkedin_username))
    })
    it("Should NOT contain the linkedin_username", () => {
      let localValues = {
        header: { ...values.header, linkedin_username: undefined }
      }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.find("ul.icons li#icons-linkedin").exists()).toBe(false)
    })

    //twitter_username
    it("Should contain the twitter_username", () => {
      const wrapper = factory({ options, values })
      expect(
        wrapper.find("ul.icons li#icons-twitter a").attributes("href")
      ).toEqual(expect.stringContaining(values.header.twitter_username))
    })
    it("Should NOT contain the twitter_username", () => {
      let localValues = {
        header: { ...values.header, twitter_username: undefined }
      }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.find("ul.icons li#icons-twitter").exists()).toBe(false)
    })

    //email
    it("Should contain the email", () => {
      const wrapper = factory({ options, values })
      expect(
        wrapper.find("ul.icons li#icons-email a").attributes("href")
      ).toEqual(expect.stringContaining(values.header.email))
    })
    it("Should NOT contain the email", () => {
      let localValues = { header: { ...values.header, email: undefined } }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.find("ul.icons li#icons-email").exists()).toBe(false)
    })

    //phone
    it("Should contain the phone", () => {
      const wrapper = factory({ options, values })
      expect(
        wrapper.find("ul.icons li#icons-phone a").attributes("href")
      ).toEqual(expect.stringContaining(values.header.phone))
    })
    it("Should NOT contain the phone", () => {
      let localValues = { header: { ...values.header, phone: undefined } }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.find("ul.icons li#icons-phone").exists()).toBe(false)
    })

    //skype_username
    it("Should contain the skype_username", () => {
      const wrapper = factory({ options, values })
      expect(
        wrapper.find("ul.icons li#icons-skype a").attributes("href")
      ).toEqual(expect.stringContaining(values.header.skype_username))
    })
    it("Should NOT contain the skype_username", () => {
      let localValues = {
        header: { ...values.header, skype_username: undefined }
      }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.find("ul.icons li#icons-skype").exists()).toBe(false)
    })
  })
})
