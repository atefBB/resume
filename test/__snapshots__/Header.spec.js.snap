// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Default Template 1`] = `
<div class="container">
  <div class="el-row">
    <div class="el-col el-col-12">
      <div class="header-left">
        <h1 id="name">
          David Turbert
        </h1>
        <h2 id="title">
          Senior Web or Lead Developer / Junior Data Scientist
        </h2>
      </div>
    </div>
    <div class="el-col el-col-12">
      <div>
        <ul class="icons">
          <li id="icons-github"><a target="_blank" href="https://www.github.com/dcyou"><i title="Github link" class="fab fa-github"></i></a></li>
          <li id="icons-linkedin"><a target="_blank" href="https://www.linkedin.com/in/dcyou"><i title="Linkedin link" class="fab fa-linkedin"></i></a></li>
          <li id="icons-twitter"><a target="_blank" href="https://twitter.com/david_turbert"><i title="Twitter link" class="fab fa-twitter"></i></a></li>
          <li id="icons-email"><a target="_blank" href="mailto:david@dcyou.fr"><i title="Email link" class="fas fa-at"></i></a></li>
          <li id="icons-phone"><a target="_blank" href="tel:+33.633.578.939"><i title="Phone link" class="fas fa-phone-alt"></i></a></li>
          <!---->
        </ul>
      </div>
    </div>
  </div>
</div>
`;
